MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs

```diff
@@ -1477,6 +1473,82 @@ private void PlatformDrawInstancedPrimitives(PrimitiveType primitiveType, int ba
             }
         }
 
+        private void PlatformGetBackBufferData<T>(Rectangle rect, T[] data, int startIndex, int count) where T : struct
```

Comment:
@Jjagg - Would it be possible to move the generic work outside this platform call?  Like could the `Marshal.SizeOf(typeof(T))` be moved to the caller and `T[] data` be converted to `byte[]`, `byte*`, or `IntPtr`?

I'm thinking from the perspective of code bloat in this method (also in the `GetData/SetData` methods in textures, vertex, and index buffers).  This is something I've noticed in the cross compiler work I've been doing... lots of template code for stuff that is largely not generic.

Either way... this isn't critical for this PR, but I would like to see about reducing the amount of generic code we generate.